### PR TITLE
Fixing couchbase tickets drop 2

### PR DESCRIPTION
**Pull request by ghost**
Fixed the following tickets in this PR:
DOC- 8619
DOC- 6484
DOC- 2670
DOC- 7492
DOC- 8234
DOC- 5765
DOC- 8175
DOC- 9106
DOC- 9107
DOC- 9109
DOC- 9111
DOC- 9122
DOC- 9123
DOC- 9419

### CODE REVIEW

File: modules/fts/pages/fts-analyzers-search-functions.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+= Analyzers - Search Functions
+
+Search functions allow users to execute full text search requests within a N1QL query. In the context of N1QL queries, a full text search index can be described as one of the following :
+* a) xref:https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/covering-indexes.html[Covering index]
+* b) Non-covering index
```
Comment (couchbase123) [inline modules/fts/pages/fts-analyzers-search-functions.adoc]: Is there a link for non-covering indexes that we should reference like Covering Indexes?

File: modules/fts/pages/fts-analyzers-search-functions.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+This characterization depends on the extent to which it could answer all aspects of the SELECT predicate and the WHERE clauses of a N1Q1 query.
```
Comment (couchbase123) [inline modules/fts/pages/fts-analyzers-search-functions.adoc]: "This characterization" -> "What index to use" ...

File: modules/fts/pages/fts-rebalance-failover.adoc [hunk 3]
```diff
@@ -5,25 +5,30 @@ The search service maintains a cluster-wide set of index definitions and metadat
 
 == Handling partitions during a rebalance
 
-The search service redistributes the index partitions across the available search service nodes for a balanced partition-node assignment during a rebalance operation. The newly assigned index partitions are built afresh over DCP feed on the new nodes. And once the new partitions are built up to the current / latest sequence numbers, they are promoted to take the live traffic, and the older partitions are deleted from the system. 
+The search service redistributes the index partitions across the available search service nodes for a balanced partition-node assignment during a rebalance operation. The newly assigned index partitions are built afresh over DCP feed on the new nodes. Once the new partitions are built up to the current / latest sequence numbers, the partitions are promoted to handle live traffic, and the older partitions are deleted from the system.
```
Comment (couchbase123) [inline modules/fts/pages/fts-rebalance-failover.adoc:9]: afresh over DCP -> "utilizing the DCP"  feed 
"built up to the current / latest sequence numbers" -> "built up to the most current or latest sequence numbers"

File: modules/fts/pages/fts-rebalance-failover.adoc [hunk 4]
```diff
@@ -5,25 +5,30 @@ The search service maintains a cluster-wide set of index definitions and metadat
...
-
+As the rebalance is a resource-consuming cluster management operation, it is always recommended to perform rebalance during off-peak hours.Search service moves or builds the index partitions one at a time per node during the rebalance operations, which could significantly increase the overall time taken for the rebalance operation.
+ 
 There is a configurable option [maxConcurrentPartitionMovesPerNode] to bring the additional concurrency to how we move/build partitions during a rebalance operation. 
-Suppose we override this parameter (maxConcurrentPartitionMovesPerNode to N) as a runtime cluster option. In that case, we could concurrently build that many partitions in parallel per node at a time, and the rebalance ought to complete faster.
+ 
+If we override this parameter(maxConcurrentPartitionMovesPerNode to N) as a runtime cluster option, then we could concurrently build that many partitions in parallel per node at a time, and the rebalance ought to complete faster.
```
Comment (couchbase123) [inline modules/fts/pages/fts-rebalance-failover.adoc]: By overriding the maxConcurrentPartitionMovesPerNode parameter using the runtime cluster option, we can control the concurrency of partition moves, enabling faster rebalance completion times.

Knowing what value to configure this parameter depends on the particular use case and available network bandwidth. Adequate testing must be done in a pre-production environment prior to deploying production clusters with this setting.

Comment (couchbase123) [reply modules/fts/pages/fts-rebalance-failover.adoc]: @sreekanth-cb - I've added some more content here re. testing in a pre-production environment and slightly made the sentence easier to read. Can you please take a look?

File: modules/fts/pages/fts-rebalance-failover.adoc [hunk 5]
```diff
@@ -39,16 +44,17 @@ Note: When multiple partitions are built parallel, it needs more RAM and mandate
 
 === Failovers 
 
-During failover of Search service nodes, there is no partition movement, and hence the failover-rebalance is instantaneous. Search service promotes the replica index partitions to primary so that those serve the live cluster traffic instantly. Users could use failover and recovery rebalance for applying patches or upgrading software/hardware for a shorter duration. 
+During failover of FTS service nodes, there is no partition movement, and hence the failover-rebalance is instantaneous. Search service promotes the replica index partitions to primary so that those serve the live cluster traffic instantly. Users could use failover and recovery rebalance for applying patches or upgrading software/hardware for a shorter duration.
```
Comment (couchbase123) [inline modules/fts/pages/fts-rebalance-failover.adoc:50]: Users could use failover and recovery rebalance for applying patches or upgrading software/hardware for a shorter duration.  -> We should rewrite this sentence. It's not clear if patch time is more straightforward or rebalance is shorter? This sentence needs a bit more simplification

File: modules/fts/pages/fts-secure-fts-queries.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+curl -XPOST ... https://<ip>:18094/api/index/{indexName}/query ...
+----
+
+The user needs to provide the username, password (-u).
+ 
+For information on using cURL in N1QL, see xref:https://blog.couchbase.com/developer-release-curl-n1ql/[blog]
+
+The curl function takes up to three parameters:
+1. The first parameter is the HTTP verb used. So far, we support “GET” and “POST.”
+2. The second parameter is the endpoint URL
```
Comment (couchbase123) [inline modules/fts/pages/fts-secure-fts-queries.adoc]: period (.) at the end of 2nd bullet

File: modules/fts/pages/fts-secure-fts-queries.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+|Basic
+|Use HTTP Basic Authentication
+|-
+
+|Insecure 
+|Allow connections to SSL sites without certs (H)
+|-
+
+|Anyauth
+|curl to figure out authentication method by itself, and use the most secure one (In our case this will be basic only for now. Eventually we will support digest)
```
Comment (couchbase123) [inline modules/fts/pages/fts-secure-fts-queries.adoc]: curl to figure out authentication method by itself, and use the most secure one. Currently, only basic auth is supported.

File: modules/fts/pages/fts-supported-queries.adoc [hunk 8]
```diff
@@ -1,4 +1,5 @@
 = Supported Queries
+:page-aliases: query-types.adoc 
 
 [abstract]
 With Full Text Search you can perform queries on Full Text Indexes. You can perform the Queries either by using Couchbase Web Console, the Couchbase REST API, N1QL (using search functions in the Query service), or the Couchbase SDK.
```
Comment (couchbase123) [inline modules/fts/pages/fts-supported-queries.adoc]: You can perform the Queries -> You can perform the queries

File: modules/fts/pages/fts-troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,244 @@
+= Troubleshooting and FAQs
+
+[abstract]
+Full Text logs can be found in the file fts.log.
+This is stored with the other Couchbase Server logs; the exact location depends on your operating system.
+
+== *Q: I just did a search but I don’t see the document contents, I just see the document IDs. I don’t want to have to click on document IDs.*
+
+You need to store fields in order to see result snippets and highlighting.
+Be aware that doing so makes your indexes larger and take longer to build, so make sure you need these features.
```
Comment (couchbase123) [inline modules/fts/pages/fts-troubleshooting.adoc]: Be aware that -> Note that

File: modules/fts/pages/fts-troubleshooting.adoc [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+== *Q: Why is indexing so slow?*
+
+By default, FTS tries to index all the fields in a document yet you typically only need to search on a few key fields.
+Indexing all fields requires more computation and results in larger indexes than you get if you limit the index to just the fields you need.
+
+Here are some tips to speed up indexing:
+
+* Build a custom mapping, set "dynamic": false to disable automatic indexing, and explicitly index only the named fields that you need to search.
+* Avoid numeric/date fields if you do not plan to query on these fields.
```
Comment (couchbase123) [inline modules/fts/pages/fts-troubleshooting.adoc]: Avoid numeric/date fields if you do not plan to query on these fields. -> Double negation avoid, not.

* Avoid querying on numeric or date fields.

File: modules/fts/pages/fts-troubleshooting.adoc [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+* Try rewriting queries to be more efficient if you must search on high frequency terms.
+Using conjunction (AND) queries can help make the search faster.
+In a query string, you can do this by using the + operator (MUST).
+For If searching for the following is slow: lowFreqTerm highFreqTerm, try +lowFreqTerm highFreqTerm.
++
+Although it is a different search, it should perform better and may be acceptable.
+
+== *Q: Why are the queries getting rejected with HTTP status code 429?*
+
+At a high level, the HTTP status code 429 indicates that ftsMemoryQuota for the cluster is under-provisioned. Whenever Search service receives a query, it tries to forecast the memory requirements for performing that search request. And the query is rejected when fts service realises that the cumulative memory usage of the service goes beyond the set memory quota limit by executing the incoming query.
```
Comment (couchbase123) [inline modules/fts/pages/fts-troubleshooting.adoc]: And the query is rejected when fts service realises that the cumulative memory usage of the service goes beyond the set memory quota limit by executing the incoming query.

When processing an incoming query, the query gets rejected by FTS when the cumulative memory usage has exceeded the set memory quota limit.

Comment (couchbase123) [reply modules/fts/pages/fts-troubleshooting.adoc]: Change "fts service" to "FTS"

File: modules/fts/pages/fts-type-mappings-add-child-field-store.adoc [hunk 12]
```diff
@@ -1,8 +1,15 @@
...
 
-When checked, the field-content is included in the set of values returned from a search; when unchecked, the field-content is not so included.
-Note that inclusion of field-content specifically permits _highlighting_ of results, so that matched expressions can be easily seen; and generally assists in debugging procedures.
-However, it also results in larger indexes and longer processing-times.
+image::fts-type-mappings-child-field-store.png[,700,align=left]
 
-// == Example
-// #Need Example here
\ No newline at end of file
+When the child field 'store' option is checked, the original field content is included in the fts index, enabling the retrieval of stored field values during a search operation.
```
Comment (couchbase123) [inline modules/fts/pages/fts-type-mappings-add-child-field-store.adoc]: fts index -> FTS index

File: modules/fts/pages/fts-type-mappings-add-child-field-store.adoc [hunk 13]
```diff
@@ -1,8 +1,15 @@
...
-Note that inclusion of field-content specifically permits _highlighting_ of results, so that matched expressions can be easily seen; and generally assists in debugging procedures.
-However, it also results in larger indexes and longer processing-times.
+image::fts-type-mappings-child-field-store.png[,700,align=left]
 
-// == Example
-// #Need Example here
\ No newline at end of file
+When the child field 'store' option is checked, the original field content is included in the fts index, enabling the retrieval of stored field values during a search operation. 
+
+When unchecked, the original field content is not included in the fts index. Storing the field within the index is necessary to support highlighting, which also needs "term vectors” for the field to be indexed.
```
Comment (couchbase123) [inline modules/fts/pages/fts-type-mappings-add-child-field-store.adoc]: fts index -> FTS index

File: modules/fts/pages/fts-rebalance-failover.adoc [hunk 14]
```diff
@@ -39,16 +44,17 @@ Note: When multiple partitions are built parallel, it needs more RAM and mandate
...
+During failover of FTS service nodes, there is no partition movement, and hence the failover-rebalance is instantaneous. Search service promotes the replica index partitions to primary so that those serve the live cluster traffic instantly. Users could use failover and recovery rebalance for applying patches or upgrading software/hardware for a shorter duration. 
  
 === Things to keep track of during a rebalance
 
-During a strict recovery rebalance operation (no extra node additions/removals), the index partitions residing on the recovered node would be reused. And this ensures a quick recovery rebalance operation.
+A failed over node can be recovered: added back into the cluster from which it was failed over through the rebalance operation. 
+The index partitions residing on the recovered node would be reused during a recovery rebalance operation with no extra node additions/removals. And this ensures a quick recovery rebalance operation.
```
Comment (couchbase123) [inline modules/fts/pages/fts-rebalance-failover.adoc]: And this ensures a quick recovery rebalance operation. -> This speeds up the recovery rebalance operation.

File: modules/fts/pages/fts-analyzers-search-functions.adoc (hunk 1 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-analyzers-search-functions.adoc]: Couldn't find any direct links in our documentation.

File: modules/fts/pages/fts-monitor.adoc [hunk 15]
```diff
@@ -1,11 +1,104 @@
...
+* Search Mutations Remaining: The number of mutations not yet indexed. 
+* Search Partitions: The number of actual index partitions.
+* Search Partitions Expected: The number of partitions expected to be scanned for the search query.
+* Search Records to Persist: The number of index records not yet persisted to the disk.
+* Search Index Rate: The rate (In Bytes per second) of indexing plain text.
+* Search Result Rate: The rate (In Bytes per second) of returning the search result.
+* Search Compaction Rate: The rate (In Bytes per second) of compaction of the index segments.
+* Search Query Rate: The number of queries run per second.
+* Search Query Error Rate: The number of queries per second (Including timeouts) that resulted in an error.
+* Search Slow Queries: The number of queries that run slowly (greater than 5 seconds to run) ) per second.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-monitor.adoc]: G in capital case for (greater...) ?
In all other cases, starting letter is in caps)

File: modules/fts/pages/fts-monitor.adoc [hunk 16]
```diff
@@ -1,11 +1,104 @@
...
+
+[source,shell]
+----
+curl -XGET -H "Content-Type: application/json" -uAdministrator:asdasd 'http://localhost:9200/api/query/index/<indexName>?longerThan=1ms
+----
+
+.Sample Output
+[source,json]
+----
+{ "status": "ok", "totalActiveQueryCount": 3, "filteredActiveQueries": { "indexName": "DemoIndex", "longerThan": "1s", "queryCount": 3, "queryMap": { "4": { "QueryContext": { "query": { "query": "ipa" }, "size": 10, "
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-monitor.adoc]: this formating/indentation looks really off. 
The problem is - it isn't readable for the user?
Ideally, this response json has to be JSON formatted.

File: modules/fts/pages/fts-multi-collection-behaviour.adoc [hunk 17]
```diff
@@ -0,0 +1,11 @@
+= Multiple Collection Behaviour
+
+Couchbase's Full Text Search service is the only Couchbase service that can create indexes that span collections.
+
+The result set of a query that references a multi-collection index returns the collection names alongside the document ID, making it easy for users to identify which collection a document belongs to.
+ 
+//ADD a query example
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-multi-collection-behaviour.adoc]: why these comments here? stale contents.

Comment (sreekanth-cb) [reply modules/fts/pages/fts-multi-collection-behaviour.adoc]: this whole content was supposed to go in abother PR.

File: modules/fts/pages/fts-quick-index.adoc [hunk 18]
```diff
@@ -27,7 +27,11 @@ image::fts-quick-index-json-configuration.png[,750,align=left]
 
 Select the related type for the field from the dropdown. Select Index this field as an identifier to index the identifier values exactly as it is without any transformation; for this case, language selection is disabled. 
 
-Configuration allows you to specify the language selection for the specified field. You can select the given configuration , but it requires additional storage and will make the index larger.
+The configuration panel allows the user to select the language for the specified field. The user can choose the default configuration, but it requires additional storage and makes the index larger.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-quick-index.adoc]: This content is still wrong. Not sure from where this is getting picked up.

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 3 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-rebalance-failover.adoc:9]: sounds good!

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 4 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-rebalance-failover.adoc]: 👍

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 14 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-rebalance-failover.adoc]: 👍

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 5 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-rebalance-failover.adoc:50]: Please feel free to propose new contents./wordings..

File: modules/fts/pages/fts-secure-fts-queries.adoc [hunk 19]
```diff
@@ -0,0 +1,44 @@
+= Secure Queries Using SSL
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-secure-fts-queries.adoc:1]: Was this supposed to be a part of this PR? IIRC, it is not as per the doc writers..

File: modules/fts/pages/fts-supported-queries.adoc (hunk 8 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-supported-queries.adoc]: 👍

File: modules/fts/pages/fts-troubleshooting.adoc (hunk 10 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-troubleshooting.adoc]: This changes the meaning intented here.

Original intent was this - user should avoid indexing numeric/date fields if they dont have query use case/requirements around those non-text fields.

**Comment by sreekanth-cb** [comment]
@couchbase123 , I am good with most of your recommendations except one for which feedbacks are given. Thanks!

File: modules/fts/pages/fts-analyzers-search-functions.adoc (hunk 2 above)
Comment (ghost) [reply modules/fts/pages/fts-analyzers-search-functions.adoc]: "This characterization" seems to be more precise.

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 3 above)
Comment (ghost) [reply modules/fts/pages/fts-rebalance-failover.adoc:9]: Updated

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 4 above)
Comment (ghost) [reply modules/fts/pages/fts-rebalance-failover.adoc]: Updated

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 5 above)
Comment (ghost) [reply modules/fts/pages/fts-rebalance-failover.adoc:50]: Users could use failover and recovery rebalance for applying patches or upgrading software/hardware for a shorter duration. -> Users can use failover and recovery rebalance while applying patches or upgrading software/hardware.

File: modules/fts/pages/fts-secure-fts-queries.adoc (hunk 6 above)
Comment (ghost) [reply modules/fts/pages/fts-secure-fts-queries.adoc]: Updated

File: modules/fts/pages/fts-secure-fts-queries.adoc (hunk 7 above)
Comment (ghost) [reply modules/fts/pages/fts-secure-fts-queries.adoc]: Updated

File: modules/fts/pages/fts-supported-queries.adoc (hunk 8 above)
Comment (ghost) [reply modules/fts/pages/fts-supported-queries.adoc]: Updated

File: modules/fts/pages/fts-troubleshooting.adoc (hunk 10 above)
Comment (ghost) [reply modules/fts/pages/fts-troubleshooting.adoc]: Updated

File: modules/fts/pages/fts-troubleshooting.adoc (hunk 11 above)
Comment (ghost) [reply modules/fts/pages/fts-troubleshooting.adoc]: Updated

File: modules/fts/pages/fts-type-mappings-add-child-field-store.adoc (hunk 12 above)
Comment (ghost) [reply modules/fts/pages/fts-type-mappings-add-child-field-store.adoc]: Updated

File: modules/fts/pages/fts-type-mappings-add-child-field-store.adoc (hunk 13 above)
Comment (ghost) [reply modules/fts/pages/fts-type-mappings-add-child-field-store.adoc]: Updated

File: modules/fts/pages/fts-rebalance-failover.adoc (hunk 14 above)
Comment (ghost) [reply modules/fts/pages/fts-rebalance-failover.adoc]: Updated

File: modules/fts/pages/fts-monitor.adoc (hunk 15 above)
Comment (ghost) [reply modules/fts/pages/fts-monitor.adoc]: Updated

File: modules/fts/pages/fts-monitor.adoc (hunk 16 above)
Comment (ghost) [reply modules/fts/pages/fts-monitor.adoc]: JSON Formatted

File: modules/fts/pages/fts-secure-fts-queries.adoc (hunk 19 above)
Comment (ghost) [reply modules/fts/pages/fts-secure-fts-queries.adoc:1]: Will be included in the next PR.

**Review by simon-dew — APPROVED** [review]

**Pull request merged**